*** y_tab.cpp	Wed Sep 10 11:19:19 2008
--- y_tab.cpp.new	Wed Sep 10 11:22:05 2008
***************
*** 9,14 ****
--- 9,26 ----
  
  #include <stdio.h>
  
+ #ifdef VERSION
+ #undef VERSION
+ #endif
+ 
+ #ifdef WIN32
+ #undef CONST
+ #undef VOID
+ #undef IN
+ #undef OUT
+ #undef OPAQUE
+ #endif
+ 
  void yyunput(int c);
  
  #if (defined(apollo) || defined(HPUX) || defined(SGI)) && defined(__cplusplus)
***************
*** 118,125 ****
  # define PRAGMA_VERSION 319
  # define PRAGMA_END 320
  
- #include <inttypes.h>
- 
  #ifdef __STDC__
  #include <stdlib.h>
  #include <string.h>
--- 118,123 ----
***************
*** 130,136 ****
--- 128,136 ----
  #define YYCONST
  #endif
  
+ #if !defined(WIN32) && !defined(__Lynx__) && !defined(__QNXNTO__) && !defined(DDS_NETBSD) && !defined(_DARWIN_C_SOURCE)
  #include <values.h>
+ #endif
  
  #if defined(__cplusplus) || defined(__STDC__)
  
***************
*** 1042,1049 ****
   * Copyright (c) 1993 by Sun Microsystems, Inc.
   */
  
- #pragma ident   "@(#)yaccpar    6.16    99/01/20 SMI"
- 
  /*
  ** Skeleton parser driver for yacc output
  */
--- 1042,1047 ----
